src/app/shared/copier/copier.service.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * This class is based on the code in the following projects:
+ *
+ * - https://github.com/zenorocha/select
+ * - https://github.com/zenorocha/clipboard.js/
+ *
+ * Both released under MIT license - © Zeno Rocha
+ */
+import {Injectable} from '@angular/core';
+
+@Injectable()
+export class CopierService {
+  private fakeElem: HTMLTextAreaElement;
```

Comment:
`fakeElem` -> `textarea`?

It's weird to call it fake, since it's a real `textarea`, just hidden